src/web/rustdoc.rs

```diff
@@ -362,18 +362,26 @@ pub(crate) async fn rustdoc_html_server_handler(
     // Pages generated by Rustdoc are not ready to be served with a CSP yet.
     csp.suppress(true);
 
+    fn get_query_and_fragment(uri: &Uri) -> Option<&str> {
```

Comment:
from what I know, the fragment will never reach the server in any case. So just using the `RawQuery` extractor and passing its content to the redirect is good enough.

Reply:
Noted!